applications/new_schema/mutations.py

```diff
@@ -83,8 +93,17 @@ def mutate_and_get_payload(cls, root, info, **input):
         application = get_node_from_global_id(
             info, input.pop("id"), only_type=WinterStorageApplicationNode
         )
+
+        if (
+            application.status != ApplicationStatus.PENDING
+            and input.get("customer_id") is None
+        ):
+            raise VenepaikkaGraphQLError(
+                _("Customer cannot be disconnected from processed applications")
+            )
+
         input["customer"] = get_node_from_global_id(
-            info, input.pop("customer_id"), only_type=ProfileNode
+            info, input.get("customer_id"), only_type=ProfileNode, nullable=True
```

Comment:
Can you replace the `input.get("customer_id")` with
```python
customer_id = input.get("customer_id")

if (application.status != ApplicationStatus.PENDING and not customer_id):
    raise ...

input["customer"] = get_node_from_global_id(info, customer_id, only_type=ProfileNode, nullable=True)
```

Reply:
Done.